src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs

```diff
@@ -143,6 +146,275 @@ impl CompactionPicker for MinOverlappingPicker {
+            for (right, table) in select_tables.iter().enumerate().skip(left) {
+                if level_handler.is_pending_compact(&table.sst_id) {
+                    break;
+                }
+
+                if all_file_size > self.max_compaction_bytes
+                    || all_file_count > self.max_file_count as usize
+                {
+                    break;
+                }
+
+                let mut select_sst_id_set = BTreeSet::default();
+                let mut level_select_files: Vec<Vec<SstableInfo>> = vec![vec![]; l0.len()];
+                level_select_files[0].extend(select_tables[left..=right].to_vec());
+                let mut overlap_info = self.overlap_strategy.create_overlap_info();
+                for sst in &level_select_files[0] {
+                    overlap_info.update(sst);
+                    select_sst_id_set.insert(sst.sst_id);
+
+                    all_file_size += sst.file_size;
+                }
+
+                all_file_count += right - left + 1;
+
+                let mut select_level_count = 1;
+                let mut last_level_index = 0;
+
+                let mut pending_compact = false;
+                for (target_index, target_level) in l0.iter().enumerate().skip(1) {
+                    if target_level.level_type() != LevelType::Nonoverlapping {
+                        break;
+                    }
+
+                    if all_file_size > self.max_compaction_bytes
+                        || all_file_count > self.max_file_count as usize
+                    {
+                        break;
+                    }
+
+                    let target_tables = &target_level.table_infos;
+                    let overlap_files = overlap_info.check_multiple_overlap(target_tables);
+
+                    for other in &overlap_files {
+                        if level_handler.is_pending_compact(&other.sst_id) {
+                            pending_compact = true;
+                            break;
+                        }
+                        overlap_info.update(other);
+                        select_sst_id_set.insert(other.sst_id);
+
+                        all_file_size += other.file_size;
+                        all_file_count += 1;
+                    }
+
+                    if pending_compact {
+                        break;
+                    }
+
+                    last_level_index = target_index;
+
+                    if overlap_files.is_empty() {
+                        // We allow a layer in the middle without overlap, so we need to continue to
+                        // the next layer to search for overlap
+                        continue;
+                    }
+
+                    level_select_files[target_index].extend(overlap_files.into_iter());
+                    select_level_count += 1;
+                }
+
+                if pending_compact {
+                    break;
+                }
+
+                // check reverse overlap
```

Comment:
As shown in your code, reverse overlap is necessary, but what about reverse-reverse overlap?